antidox/tester.py

```diff
@@ -0,0 +1,79 @@
+""" Program that's gets the latest revisions and checks for PII and toxicity """
+import argparse
+import json
+import os
+import time
+import datetime
+import requests
+import pywikibot
+from antidox import clean
+from antidox import perspective
+
+
+def wiki_write(result, header):
+  """ Writes results to wikipedia/wikimedia bot page """
+  pywikibot.config.register_family_file('doxwiki', os.path.join(os.path.dirname(__file__), 'doxwiki_family.py'))
+  pywikibot.config.usernames['doxwiki']['en'] = u'Antidoxer'
+  site = pywikibot.Site()
+  repo = site.data_repository()
+  page = pywikibot.Page(site, u"User_talk:Antidoxer")
+
+  heading = (header)
+  content = (result)
+  message = "\n\n{}\n{} --~~~~".format(heading, content)
+  page.save(summary="Testing", watch=None, minor=False, botflag=True,
+            force=False, async=False, callback=None,
+            apply_cosmetic_changes=None, appendtext=message)
+
+def log_change():
+  """ gets latest revisions and cleans them """
+  while True:
+    start = datetime.datetime.now() - datetime.timedelta(minutes=2)
+    rcstart = start.isoformat()
+    page = (args.mediawiki + "api.php?action=query&list=recentchanges&rclimit=500&rcprop=title%7Cids%7Csizes%7Cflags%7Cuser&rcdir=newer&rcstart="+ rcstart + "&rcend=now&format=json")
+    get_page = requests.get(page)
+    response = json.loads(get_page.content)
+
+    for change in response['query']['recentchanges']:
+      print('new change:')
+      revid = str(change['revid'])
+      old_revid = str(change['old_revid'])
+      compare = (args.mediawiki + "api.php?action=compare&fromrev="
+                 + old_revid + "&torev=" + revid + "&format=json")
+      get_compare = requests.get(compare)
+      response = json.loads(get_compare.content.decode('utf-8'))
+
+      if 'compare' not in response:
+        continue
+      revision = response['compare']['*']
+      text = clean.content_clean(revision)
+
+      apikey_data, toxicity, dlp = perspective.get_client()
```

Comment:
We're repeatedly getting new Perspective and DLP client objects every time, which is unnecessary. You can move this outside/above the `while` loop so that this step only happens once.